front/app/components/Form/Components/Layouts/utils.ts

```diff
@@ -0,0 +1,90 @@
+import {
+  and,
+  isCategorization,
+  JsonSchema,
+  Layout,
+  Tester,
+  UISchemaElement,
+  uiTypeIs,
+} from '@jsonforms/core';
+import { forOwn, isEmpty } from 'lodash-es';
+
+export interface PageType extends Layout {
+  type: 'Page';
+  /**
+   * The label associated with this category layout.
+   */
+  label: string;
+}
+
+/**
+ * The categorization element, which may have children elements.
+ * A child element may either be itself a Categorization or a Page, hence
```

Comment:
It is good to support nested elements, because it allows grouping of fields, which is still used in the idea form. But we will never nest a page inside a page. Functionally it would not makes sense, would it?